Simple Caddyfile crashes after #6997

After #6997 was merged, Caddy built from the `master` branch crashes after running

``` shell
caddy run --config Caddyfile
```

with the following Caddyfile:

``` Caddyfile
{
    skip_install_trust
}
```

(This obviously isn't a useful Caddyfile - it's the result of me paring down something more useful.)

I'll include the console output at the bottom of this message, as it's long.

I'm running macOS 15.4.1 (ARM CPU) and building Caddy with Go 1.24.3 and xcaddy 0.4.4 by running

``` shell
xcaddy build 320c57291dbe06e00e0759bdb5cbbf0d622e5968
```

There's no crash when building the previous commit:

``` shell
xcaddy build aa3d20be3ee451af9465470a28937690104e9422
```

Here's the console output mentioned earlier:

``` text
{"level":"info","ts":1746589905.9815402,"msg":"maxprocs: Leaving GOMAXPROCS=10: CPU quota undefined"}
{"level":"info","ts":1746589905.981635,"msg":"GOMEMLIMIT is updated","package":"github.com/KimMachineGun/automemlimit/memlimit","GOMEMLIMIT":61847529062,"previous":9223372036854775807}
{"level":"info","ts":1746589905.981685,"msg":"using config from file","file":"Caddyfile"}
{"level":"info","ts":1746589905.9818199,"msg":"adapted config to JSON","adapter":"caddyfile"}
{"level":"info","ts":1746589905.9820008,"msg":"maxprocs: No GOMAXPROCS change to reset"}
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/caddyserver/caddy/v2.Context.LoadModuleByID({{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, ...}, ...)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/context.go:404 +0x594
github.com/caddyserver/caddy/v2.Context.App({{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, ...}, ...)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/context.go:494 +0x148
github.com/caddyserver/caddy/v2.Context.AppIfConfigured({{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, ...}, ...)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/context.go:521 +0x190
github.com/caddyserver/caddy/v2/modules/caddypki.(*adminAPI).Provision(0x1400047d680, {{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/modules/caddypki/adminapi.go:56 +0x170
github.com/caddyserver/caddy/v2.(*AdminConfig).provisionAdminRouters(0x14000384e00, {{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/admin.go:292 +0xec
github.com/caddyserver/caddy/v2.replaceLocalAdminServer(0x140003849a0, {{0x1026e2a48, 0x14000165720}, 0x140004ff7d0, 0x140003849a0, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/admin.go:429 +0x224
github.com/caddyserver/caddy/v2.provisionContext(0x1400053b298?, 0x1)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/caddy.go:510 +0x24c
github.com/caddyserver/caddy/v2.run(0x14000060870?, 0x1)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/caddy.go:401 +0x40
github.com/caddyserver/caddy/v2.unsyncedDecodeAndRun({0x140000607d0, 0x4e, 0x50}, 0x1)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/caddy.go:344 +0xe4
github.com/caddyserver/caddy/v2.changeConfig({0x101ddbfca, 0x4}, {0x101de2ea6, 0x7}, {0x14000060780, 0x4e, 0x50}, {0x0, 0x0}, 0x1)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/caddy.go:235 +0x5b8
github.com/caddyserver/caddy/v2.Load({0x14000060780?, 0x9?, 0x0?}, 0x0?)
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/caddy.go:134 +0x88
github.com/caddyserver/caddy/v2/cmd.cmdRun({0x0?})
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/cmd/commandfuncs.go:239 +0x5dc
github.com/caddyserver/caddy/v2/cmd.init.1.func2.WrapCommandFuncForCobra.1(0x140005fe008, {0x101ddc026?, 0x4?, 0x101ddbffa?})
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/cmd/cobra.go:141 +0x34
github.com/spf13/cobra.(*Command).execute(0x140005fe008, {0x140000b34c0, 0x2, 0x2})
    github.com/spf13/cobra@v1.9.1/command.go:1015 +0x844
github.com/spf13/cobra.(*Command).ExecuteC(0x1400021b808)
    github.com/spf13/cobra@v1.9.1/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(...)
    github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/caddyserver/caddy/v2/cmd.Main()
    github.com/caddyserver/caddy/v2@v2.10.1-0.20250506212838-320c57291dbe/cmd/main.go:72 +0x6c
main.main()
    caddy/main.go:11 +0x1c
```

--- a/caddy.go
+++ b/caddy.go
@@ -505,14 +505,6 @@ func provisionContext(newCfg *Config, replaceAdminServer bool) (Context, error) 
 		return ctx, err
 	}
 
-	// start the admin endpoint (and stop any prior one)
-	if replaceAdminServer {
-		err = replaceLocalAdminServer(newCfg, ctx)
-		if err != nil {
-			return ctx, fmt.Errorf("starting caddy administration endpoint: %v", err)
-		}
-	}
-
 	// create the new filesystem map
 	newCfg.fileSystems = &filesystems.FileSystemMap{}
 
@@ -542,6 +534,14 @@ func provisionContext(newCfg *Config, replaceAdminServer bool) (Context, error) 
 	}()
 	if err != nil {
 		return ctx, err
+	}
+
+	// start the admin endpoint (and stop any prior one)
+	if replaceAdminServer {
+		err = replaceLocalAdminServer(newCfg, ctx)
+		if err != nil {
+			return ctx, fmt.Errorf("starting caddy administration endpoint: %v", err)
+		}
 	}
 
 	// Load and Provision each app and their submodules

